dspace-spring-rest/src/main/java/org/dspace/app/rest/repository/CommunityRestRepository.java

```diff
@@ -96,6 +96,10 @@ public CommunityRest findOne(Context context, UUID id) {
     // with pagination and authorization check
     @SearchRestMethod(name = "subCommunities")
     public Page<CommunityRest> findSubCommunities(@Param(value = "parent") UUID parentCommunity, Pageable pageable) {
+        if (parentCommunity == null) {
```

Comment:
another option here could be to create an extension of the Param annotation MandatoryParam or RequiredParam and introduce the check in org.dspace.app.rest.utils.RestRepositoryUtils.executeQueryMethod(DSpaceRestRepository, MultiValueMap<String, Object>, Method, Pageable, Sort, PagedResourcesAssembler) throwing a specific exception. What do you think?

Reply:
@abollini , 

I think it makes sense to have explicit error handling for missing parameters.  

I will throw a more specific exception when this parameter is not present.  

I am not used to programming with annotations, so that is not how I normally think to solve a problem. 

I am finding the layering of annotations in the new rest api confusing.  Is your suggestion of a custom annotation a best practice when working with spring?

Reply:
@abollini , could we use the same mechanism as is provided in the following signature to designate required parameters: https://github.com/DSpace/DSpace/blob/master/dspace-spring-rest/src/main/java/org/dspace/app/rest/AuthenticationRestController.java#L88-L89